actors/evm/src/interpreter/system.rs

```diff
@@ -385,6 +434,49 @@ impl<'r, RT: Runtime> System<'r, RT> {
         Ok(())
     }
 
+    /// Returns the current transient data lifespan based on the execution environment.
+    pub fn get_current_transient_data_lifespan(rt: &RT) -> Option<TransientDataLifespan> {
+        match rt.message().origin().id() {
+            Ok(origin_id) => {
+                Some(TransientDataLifespan { origin: origin_id, nonce: rt.message().nonce() })
```

Comment:
I'd consider storing this in the `System` object. It'll be the same over the course of the transaction.

Reply:
a47c21e36338f087a92762a8bb3cc48acc2b8e99